parsing/location.ml

```diff
@@ -423,10 +438,8 @@ let () =
       | Sys_error msg ->
           Some (errorf ~loc:(in_file !input_name)
                 "I/O error: %s" msg)
-      | Warnings.Errors n ->
-          Some
-            (errorf ~loc:(in_file !input_name)
-             "Some fatal warnings were triggered (%d occurrences)" n)
+
+      | Warnings.Errors _ -> Some non_displayed_error
```

Comment:
I started this way but there are two things that were a bit awkward:
- it doesn't allow to define more exceptions that don't need to be displayed
- Location.error_of_exn is exposed

ocamldoc can receive Warnings.Error but it uses Location.error_of_exn for instance, so it's report some weird errors. But in fact, it doesn't even report warnings in mli because it's missing a call to Typecore.force_delayed_checks, and it doesn't stop on fatal warnings in ml because it's missing a call to Warnings.check_fatal (why is it even reporting warnings anyway?).

Looking at the uses of Location.error_of_exn, I think they are all worse than using Location.report_exception, so I guess I'll change that, and then it'd make sense to do what you say.